### PR TITLE
Added url search parameters for insecure e2e testing

### PR DESCRIPTION
**Pull request by shrshindeMSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

Added search URL parameter `origins` to send origins to `app.initialize()`.

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

1. Added a URL search parameter `origins`.

### Validation performed:

N/A

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

No
### End-to-end tests added:

No

### CODE REVIEW

File: apps/teams-test-app/src/App.tsx [hunk 1]
```diff
@@ -68,12 +68,18 @@ import WebStorageAPIs from './components/WebStorageAPIs';
 
 const urlParams = new URLSearchParams(window.location.search);
 
+const getOriginParam =
+  urlParams.has('doesInsecureUrlHasAppIdException') && urlParams.get('doesInsecureUrlHasAppIdException')
```
Comment (erinha) [inline apps/teams-test-app/src/App.tsx:72]:
```suggestion
  urlParams.has('doesInsecureUrlHaveAppIdException') && urlParams.get('doesInsecureUrlHaveAppIdException')
```

**Comment by erinha** [comment]
Can you please update the PR description too (after the param rename)?

File: apps/teams-test-app/src/App.tsx (hunk 1 above)
Comment (shrshindeMSFT) [reply apps/teams-test-app/src/App.tsx]: Updated the name to `origins` it sounded better

File: apps/teams-test-app/src/App.tsx [hunk 2]
```diff
@@ -68,12 +68,15 @@ import WebStorageAPIs from './components/WebStorageAPIs';
 
 const urlParams = new URLSearchParams(window.location.search);
 
+const getOriginParam = urlParams.has('origins') && urlParams.get('origins') ? urlParams.get('origins') : '';
```
Comment (lakhveerk) [inline apps/teams-test-app/src/App.tsx:71]: can this be turned into 
```const getOriginParam = urlParams.get('origins') || '';```

OR ```const getOriginParam = urlParams.has('origins') ? urlParams.get('origins') || '' : '';```
If you have to call `.has` as well

File: apps/teams-test-app/src/App.tsx [hunk 3]
```diff
@@ -68,12 +68,15 @@ import WebStorageAPIs from './components/WebStorageAPIs';
...
+const validMessageOrigins: string[] = getOriginParam ? [getOriginParam] : [];
+
 // This is added for custom initialization when app can be initialized based upon a trigger/click.
 if (!urlParams.has('customInit') || !urlParams.get('customInit')) {
   if (isTestBackCompat()) {
-    initialize();
+    initialize(undefined, validMessageOrigins);
   } else {
-    app.initialize();
+    app.initialize(validMessageOrigins);
```
Comment (erinha) [inline apps/teams-test-app/src/App.tsx:79]: Should it be passing empty array or undefined if `getOriginParam` is not specified? Maintaining the previous behavior if it's not being overridden would be passing undefined.

File: apps/teams-test-app/src/App.tsx (hunk 2 above)
Comment (erinha) [inline apps/teams-test-app/src/App.tsx:71]: nit
```suggestion
const getOriginsParam = urlParams.has('origins') && urlParams.get('origins') ? urlParams.get('origins') : '';
```

File: apps/teams-test-app/src/App.tsx [hunk 4]
```diff
@@ -68,12 +68,15 @@ import WebStorageAPIs from './components/WebStorageAPIs';
 
 const urlParams = new URLSearchParams(window.location.search);
 
+const getOriginParam = urlParams.has('origins') && urlParams.get('origins') ? urlParams.get('origins') : '';
+const validMessageOrigins: string[] = getOriginParam ? [getOriginParam] : [];
```
Comment (lakhveerk) [inline apps/teams-test-app/src/App.tsx:72]: I just learned that empty string is considered falsy in javascript 📖

File: apps/teams-test-app/src/App.tsx (hunk 3 above)
Comment (shrshindeMSFT) [reply apps/teams-test-app/src/App.tsx:79]: The check on line 72 ensures we pass in the correct `validMessageOrigins`

File: apps/teams-test-app/src/App.tsx (hunk 2 above)
Comment (shrshindeMSFT) [reply apps/teams-test-app/src/App.tsx]: Updated

File: apps/teams-test-app/src/App.tsx [hunk 5]
```diff
@@ -68,12 +68,15 @@ import WebStorageAPIs from './components/WebStorageAPIs';
 
 const urlParams = new URLSearchParams(window.location.search);
```
Comment (erinha) [inline apps/teams-test-app/src/App.tsx:70]: Can you add a comment outlining how to craft your URL search parameters so they can be succesfully parsed here please?

Comment (shrshindeMSFT) [reply apps/teams-test-app/src/App.tsx:70]: Okay I've added the comment here.

**Review by erinha — APPROVED** [review]
🚀

**Review by jekloudaMSFT — APPROVED** [review]

**Pull request merged**